src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,83 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { useDispatch, useSelector } from 'react-redux';
+import { RootState } from '../../app/store';
+import { clearCurrentTodo } from '../../features/currentTodo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
 
 export const TodoModal: React.FC = () => {
+  const dispatch = useDispatch();
+  const currentTodo = useSelector((state: RootState) => state.currentTodo);
+  const { id, title, completed, userId } = currentTodo!;
+  const [userInfo, setUserInfo] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+
+  useEffect(() => {
+    const findUser = async (newUser: number) => {
+      setIsLoading(true);
+      try {
+        const user = await getUser(newUser);
+
+        setUserInfo(user);
+      } catch {
+        dispatch(clearCurrentTodo());
+        alert(`Didn't find user info`);
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    findUser(userId);
+  }, [userId, dispatch]);
+
+  const handleClickCloseModal = () => {
+    dispatch(clearCurrentTodo());
+  };
+
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      <Loader />
-
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
+      {isLoading ? (
+        <Loader />
+      ) : (
+        <div className="modal-card">
+          <header className="modal-card-head">
+            <div
+              className="modal-card-title has-text-weight-medium"
+              data-cy="modal-header"
+            >
+              Todo #{id}
+            </div>
 
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button type="button" className="delete" data-cy="modal-close" />
-        </header>
+            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+            <button
+              type="button"
+              className="delete"
+              data-cy="modal-close"
+              onClick={() => handleClickCloseModal()}
```

Comment:
You don't need to create an anonymous function here. You can directly pass the `handleClickCloseModal` function as the onClick handler.